src/main/java/org/traccar/helper/IpRetriever.java

```diff
@@ -0,0 +1,50 @@
+package org.traccar.helper;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Gets the client's IP address regardless of whether
```

Comment:
Please remove javadoc comments from everywhere. We only comment code where it's absolutely necessary.

Reply:
Okay but just out of curiosity, why do you apply such policy?

Reply:
My opinion is that code should be self-documenting. Adding comments adds extra maintenance. I think it's pretty general consensus nowadays that if your code requires a lot of comments, you are doing something wrong. One possible exception is frameworks/libraries javadoc where you need to generate API documentations. And even in that case, usually only public methods have comments.